<?xml version="1.0"?>
<launch>

    <!-- NOTE : This is an example launch file, please do not modify! -->

    <arg name="base_path" default="$(find mcr_pddl_problem_generator)/ros/test/example_domain" />
    <arg name="domain_path" default="$(arg base_path)/domain.pddl" />
    <arg name="problem_path" default="$(arg base_path)/problems/p01.pddl" />
    <arg name="cost_required" default="true" />
    <arg name="cost_file_1" default="$(arg base_path)/costs/cost_example_1.pddl" if="$(arg cost_required)" />
    <arg name="cost_file_2" default="$(arg base_path)/costs/cost_example_2.pddl" if="$(arg cost_required)" />
    <arg name="max_goals" default="4" />

    <!-- automatic PDDL problem generator node from knowledge base snapshot -->
    <node pkg="mcr_pddl_problem_generator" type="pddl_problem_generator_node" name="pddl_problem_generator_node" output="screen" ns="mcr_pddl_problem_generator" >
        <param name="domain_path" value="$(arg domain_path)" />
        <param name="problem_path" value="$(arg problem_path)" />
        <param name="max_goals" value="$(arg max_goals)" />
        <rosparam param="cost_file_paths" subst_value="True" if="$(arg cost_required)" >
            [$(arg cost_file_1), $(arg cost_file_2)]</rosparam>
    </node>

</launch>
